### DISCUSSION TITLE
How do I acquire a token, then use it to upload to endpoint?

### DISCUSSION THREAD

**Discussion by geekgarious** [thunderclient/thunder-client-support#860] [Q&A]
I've started to work on an AWS project at my firm, and have been asked to test a MuleSoft endpoint.  This requires that I use a client ID and secret to obtain a bearer token, then use this token to log in and upload data.  My co-workers are using Postman to do this, but I am a person who is blind and Postman does not seem accessible.  I found Thunder Client which does seem accessible since it is a VS Code extension.  How can I use the client to perform what is being asked?

Thanks in advance.
Eric

**Comment by rangav** [comment]
@geekgarious you can OAuth tab, see image below, set the value in red and generate token

<img width="1347" alt="Screenshot 2022-11-01 at 18 27 05" src="https://user-images.githubusercontent.com/8637550/199310074-fa1cf3ae-107b-46ce-8355-679980271c19.png">

Let me know if you can generate token or not?

**Reply by rangav** [reply]
or you can ask your mates how are they generating token in postman

**Comment by geekgarious** [comment]
I cannot read screenshots.  My screen-reader just says "image".  Could you describe the steps as text?

In order to perform my test, I need a client that supports API key authentication.

**Reply by rangav** [reply]
where you need to add api key? is it to header or query parameter

**Reply by geekgarious** [reply]
Header

**Reply by rangav** [reply]
Ok you can use headers tab then
